ragna/core/_rag.py

```diff
@@ -188,7 +202,14 @@ async def prepare(self) -> Message:
                 detail=RagnaException.EVENT,
             )
 
-        await self._run(self.source_storage.store, self.documents)
+        if list[Document] in inspect.signature(self.source_storage.store).parameters.values():
```

Comment:
1. We should have this kind of logic on as class attribute on the `SourceStorage` itself. Otherwise, how are we going to communicate this to the REST API / web UI? This needs to be known, because it makes no sense to force the user to select an embedding model when it is unused by the backend.
2. This check needs to be more strict. We should only check the first argument rather than the whole signature.